rust-src/wallet_library/src/statement.rs

```diff
@@ -0,0 +1,375 @@
+        &self,
+        config: &WalletConfig<C, AttributeType>,
+    ) -> AcceptableRequestResult {
+        match self {
+            CredentialStatement::Account {
+                statement,
+                network: _,
+                cred_id: _,
+            } => {
+                if statement.is_empty() {
+                    return Err(RequestCheckError::EmptyCredentialStatement);
+                }
+                let mut used_tags = HashSet::<AttributeTag>::new();
+                for atomic_statement in statement {
+                    let attribute = atomic_statement.attribute();
+                    if used_tags.contains(&attribute) {
+                        return Err(RequestCheckError::DuplicateTag);
+                    }
+                    used_tags.insert(attribute);
+
+                    atomic_statement.acceptable_atomic_statement(&config.identity_rules)?
+                }
+                Ok(())
+            }
+            CredentialStatement::Web3Id {
+                ty: _,
+                network: _,
+                contract: _,
+                credential: _,
+                statement: _,
+            } => Err(RequestCheckError::Web3IdStatementNotSupported),
+        }
+    }
+}
+
+// Note that this is an implementation only for the account statement, the tag
+// is AttributeTag.
+impl<C: Curve, TagType: AttributeTagType, AttributeType: Attribute<C::Scalar>>
+    AcceptableAtomicStatement<C, TagType, AttributeType>
+    for AtomicStatement<C, TagType, AttributeType>
+{
+    fn acceptable_atomic_statement(
+        &self,
+        config_rules: &Option<WalletConfigRules<C, TagType, AttributeType>>,
+    ) -> AcceptableRequestResult {
+        // Simple checks
+        match self {
+            AtomicStatement::RevealAttribute { statement: _ } => return Ok(()),
+            AtomicStatement::AttributeInRange { statement } => {
+                if statement.lower >= statement.upper {
+                    return Err(RequestCheckError::RangeMinMaxError);
+                }
+            }
+            AtomicStatement::AttributeInSet { statement } => {
+                if statement.set.is_empty() {
+                    return Err(RequestCheckError::EmptySet);
+                }
+            }
+            AtomicStatement::AttributeNotInSet { statement } => {
+                if statement.set.is_empty() {
+                    return Err(RequestCheckError::EmptySet);
+                }
+            }
+        }
+        // checks based on wallet config
+        if let Some(rules) = config_rules {
+            let check = &rules.attribute_check;
+            match self {
+                AtomicStatement::RevealAttribute { statement: _ } => (),
+                AtomicStatement::AttributeInRange { statement } => {
+                    let tag = &statement.attribute_tag;
+                    if !rules.range_tags.contains(tag) {
+                        return Err(RequestCheckError::IllegalRangeTag(format!("{}", tag)));
```

Comment:
```suggestion
                        return Err(RequestCheckError::IllegalRangeTag(tag.to_string()));
```
I'm surprised clippy does not complain :/

Reply:
Because formatting is always cool :sunglasses: (Fixed it)